lib/screens/admin_screen.rb

```diff
@@ -0,0 +1,119 @@
+    choice = 0
+    while choice != 6
+      puts "Admin Menu"
+      puts "-----------------------------------------"
+      puts "1. Add a New Book"
+      puts "2. Restock a Book"
+      puts "3. Soft Delete a Book"
+      puts "4. Show All Books"
+      puts "5. Logout"
+      puts "6. Exit"
+      print "Enter your choice: "
+
+      choice = gets.chomp.to_i
+      display_admin_message
+      case choice
+      when 1
+        add_new_book
+      when 2
+        restock_book
+      when 3
+        soft_delete_book
+      when 4
+        show_all_books
+      when 5
+        current_user = nil
+        puts "Logged out successfully!\n"
+        break
+      when 6
+        puts "Goodbye! Exiting the Library Management System."
+        exit(0)
+      else
+        puts "Invalid choice. Please try again."
+      end
+    end
+  end
+
+  def add_new_book
+    puts "Add a New Book"
+    puts "-----------------------------------------"
+    title = BookHelper.parseTitle
+    genre = BookHelper.parseGenre
+    author = BookHelper.parseAuthor
+    publish_date = BookHelper.parsePublishdate
+    count = BookHelper.parseCount
+    BookController.create(title: title, genre: genre, author: author, publish_date: publish_date, count: count)
+    puts "New book '#{title}' has been added to the inventory."
+    puts "Press enter to go back to the admin menu."
+    gets
+  end
+
+  def restock_book
+    puts "Restock a Book"
+    puts "-----------------------------------------"
+    print "Enter the id of the book you want to restock : "
+    book_id = BookHelper.parseBookid
+    book = BookController.find_by_id(book_id)
+    if book.nil?
+      puts "Book with ID #{book_id} not found."
+    else
+      print "Enter the quantity to add : "
+      quantity_to_add = gets.chomp.to_i
+      if quantity_to_add <= 0
+        puts "Quantity must be a positive number."
+      else
+        new_count = book.count.to_i + quantity_to_add
+        BookController.restock(book, new_count)
+        puts "Book '#{book.title}' has been restocked.\n"
+      end
+    end
+    puts "Press enter to go back to the admin menu."
+    gets
+  end
+
+  def soft_delete_book
+    puts "Delete a Book"
+    puts "-----------------------------------------"
+    print "Enter the id of the book you want to delete : "
+    book_id = BookHelper.parseBookid
+    book = BookController.find_by_id(book_id)
+    if book.nil?
+      puts "Book with ID #{book_id} not found."
+    else
+      BookController.soft_delete(book_id)
+      puts "Book '#{book.title}' has been soft deleted."
+    end
+    puts "Press enter to go back to the admin menu."
+    gets
+  end
+
+  def show_all_books
+    puts "Available Books : "
+    books = BookController.list_inventory
+    if books.empty?
+      puts "No books available in the inventory."
+    else
+      puts "----------------------------------------------------------------"
+      puts "ID\tTitle\tGenre\tAuthor\tPublish-Date\tCount"
```

Comment:
to display nicely in a tabular format, check how we can format this using sprintf like we do in C.